cpp/src/strings/convert/convert_hex.cu

```diff
@@ -0,0 +1,163 @@
+    }
+};
+
+/**
+ * @brief The dispatch functions for converting strings to integers.
+ *
+ * The output_column is expected to be one of the integer types only.
+ */
+struct dispatch_hex_to_integers_fn
+{
+    template <typename IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value>* = nullptr>
+    void operator()( column_device_view const& strings_column,
+                     mutable_column_view& output_column,
+                     cudaStream_t stream ) const
+    {
+        auto d_results = output_column.data<IntegerType>();
+        thrust::transform( rmm::exec_policy(stream)->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_column.size()),
+            d_results, hex_to_integer_fn<IntegerType>{strings_column});
+    }
+    // non-integral types throw an exception
+    template <typename T, std::enable_if_t<not std::is_integral<T>::value>* = nullptr>
+    void operator()(column_device_view const&, mutable_column_view&, cudaStream_t) const
+    {
+        CUDF_FAIL("Output for hex_to_integers must be an integral type.");
+    }
+};
+
+template <>
+void dispatch_hex_to_integers_fn::operator()<experimental::bool8>(column_device_view const&, mutable_column_view&, cudaStream_t) const
+{
+    CUDF_FAIL("Output for hex_to_integers must not be a boolean type.");
+}
+
+} // namespace
+
+
+// This will convert a strings column into any integer column type.
+std::unique_ptr<column> hex_to_integers( strings_column_view const& strings,
+                                         data_type output_type,
+                                         rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                         cudaStream_t stream = 0)
+{
+    size_type strings_count = strings.size();
+    if( strings_count == 0 )
+        return make_numeric_column( output_type, 0 );
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_strings = *strings_column;
+    // create integer output column copying the strings null-mask
+    auto results = make_numeric_column( output_type, strings_count,
+        copy_bitmask( strings.parent(), stream, mr), strings.null_count(), stream, mr);
+    auto results_view = results->mutable_view();
+    // fill output column with integers
+    experimental::type_dispatcher( output_type, dispatch_hex_to_integers_fn{},
+                                   d_strings, results_view, stream );
+    results->set_null_count(strings.null_count());
+    return results;
+}
+
+} // namespace detail
+
+// external API
+std::unique_ptr<column> hex_to_integers( strings_column_view const& strings,
+                                         data_type output_type,
+                                         rmm::mr::device_memory_resource* mr)
+{
+    return detail::hex_to_integers(strings, output_type, mr );
```

Comment:
Detail API should take a stream.

Reply:
This is the external API as specified in the comment at line 154.
The detail API called at line 159 is at line 131: https://github.com/rapidsai/cudf/blob/2681125332260782f8e47950f98441e783847199/cpp/src/strings/convert/convert_hex.cu#L134
It has a defaulted stream parameter.

Reply:
My mistake again. :)